src/sysext/sysext.c

```diff
@@ -848,6 +854,83 @@ static int overlayfs_paths_new(const char *hierarchy, const char *workspace_path
         return 0;
 }
 
+static int resolved_paths_equal(const char *resolved_a, const char *resolved_b) {
+        /* Returns true if paths are of the same entry, false if not, <0 on error. */
+
+        if (path_equal(resolved_a, resolved_b))
+                return 1;
+
+        if (!resolved_a || !resolved_b)
+                return 0;
+
+        return inode_same(resolved_a, resolved_b, 0);
+}
```

Comment:
This seems iffy. We already have `path_equal_or_inode_same()`, why is this function needed?

Reply:
I added it because `path_equal_or_inode_same` returns `bool`, thus swallows errors.

Reply:
can you move this into generic code, and make path_equal_or_inode_same() just a wrapper around it that eats up the error or so?